pagination/header.go

```diff
@@ -15,35 +15,54 @@ func header(u *url.URL, rel string, limit, offset int) string {
 	return fmt.Sprintf("<%s>; rel=\"%s\"", u.String(), rel)
 }
 
-func Header(u *url.URL, count int, limit, offset int) http.Header {
-	lastOffset := count + (limit - count%limit) - limit
+// Header returns an http header map, with a single key, "Link". The "Link" header is used in pagination where backwards compatibility is required.
+// The Link header will contain links any combination of the first, last, next, or previous (prev) pages in a paginated list (given a limit and an offset, and optionally a total).
+// If total is not set, then no "last" page will be calculated.
+// If no limit is provided, then this function will return an empty map.
+func Header(u *url.URL, total int, limit, offset int) http.Header {
+	if limit == 0 {
```

Comment:
Maybe just show prev/next if limit is 0?

Reply:
I think we could set a default limit (10?) or we could use the offset as the default limit.

what do you think?

Reply:
Sorry, I missread, I'd simply put limit to 1 if the value is 0 or negative!

Reply:
Alternatively only provide `first`?

Reply:
I like the idea of setting it to 1.